app/Helpers/database/user-relationship.php

```diff
@@ -71,69 +71,70 @@ function changeFriendStatus(User $senderUser, User $targetUser, int $newStatus):
     }
 }
 
-function isUserBlocking(string $user, ?string $possibly_blocked_user): bool
+function GetFriendList(User $user): array
```

Comment:
Note: this executes roughly 30ms slower than the original SQL version.

Reply:
With only four friends, my query time changed from 260us to 750us.

Using someone following 1800 people (which seems to be the max in prod right now, and makes the following page unresponsive), my query time actually went down (from 90ms to 21ms).

Additionally, as best as I can tell, this is only called from `dorequest.php?r=getfriendslist`, which I don't believe is exposed through rcheevos, so it should only be getting called if the user navigates to the friends page of the DLL overlay.

Unless I'm missing something, I don't think this is something to worry about.

Reply:
You might be running in to whatever caused [this comment](https://github.com/RetroAchievements/RAWeb/pull/2380/files#diff-08ba1639618d6064911bf201eb28ef06ea28bf0ed974101a116d4d716da9b5dfR132-R135) that would affect other relations functions: 
```
    // TODO: why is it so much faster to run this query and build the IN list
    //       than to use it as a subquery? i.e. "AND aw.User IN ($friendsSubquery)"
    //       local testing took over 2 seconds with the subquery and < 0.01 seconds
    //       total for two separate queries
```

Not sure why I'm not seeing that behavior.

Reply:
So strange, thanks for sharing your findings. The user I tested against was the same one who is following ~1800 people. My response time went up from 90ms to 130ms. Since this is only called from the Connect API under rare circumstances, you're likely right that this is nothing worth worrying about.